src/sys/socket/mod.rs

```diff
@@ -1774,7 +1774,7 @@ pub fn getsockname(fd: RawFd) -> Result<SockAddr> {
 /// In C this would usually be done by casting.  The `len` argument
 /// should be the number of bytes in the `sockaddr_storage` that are actually
 /// allocated and valid.  It must be at least as large as all the useful parts
-/// of the structure.  Note that in the case of a `sockaddr_un`, `len` need not
+/// of the structure.  Note that in the case of a `sockaddr_un`, `len` should
```

Comment:
Sure, the only thing with that is it would basically just ignore the `len` argument and do `strlen(sun_path.as_ptr(), sun_path.len())`. Unless it should *just* check?:
```rust
if sun_path.get(len) == Some(b'\0') {
    len += 1
} else {
    assert!(sun_path[len - 1] == b'0'); // maybe?
}
```

Reply:
Alright, I think I fixed it.

Reply:
What exactly did you change?  It's hard to tell since you force-pushed and github forgets about your older commits.

Reply:
In `UnixAddr::from_raw_parts`, it increments len by 1 if `path[len - 1] != 0`